tests/unit/core/test_check_updates.py

```diff
@@ -103,3 +107,36 @@ def test_check_for_updates_does_nothing_if_cache_cant_be_saved(
     latest_version = check_for_updates()
     assert latest_version is None
     request_get_mock.assert_not_called()
+
+
+def test_load_last_check_time_ok(fs: FakeFilesystem):
+    """
+    GIVEN an update cache file
+    WHEN load_last_check_time() is called
+    THEN it correctly reads the last check time
+    """
+    Path(CACHE_FILE).parent.mkdir(parents=True, exist_ok=False)
+    Path(CACHE_FILE).write_text("check-at: 2")
+
+    assert load_last_check_time() == 2
+
+
+@pytest.mark.parametrize(
+    "content",
+    [
+        "this-is-not-yaml",
+        "no-cache-key: 1",
+        "check-at: not_a_float",
+    ],
+)
+def test_load_last_check_time_swallows_errors(fs: FakeFilesystem, content: str):
+    """
+    GIVEN invalid content for the cache file
+    WHEN load_last_check_time() is called
+    THEN it does not raise any exception
+    AND it returns None
+    """
+    Path(CACHE_FILE).parent.mkdir(parents=True, exist_ok=False)
+    Path(CACHE_FILE).write_text(content)
+
+    assert load_last_check_time() is None
```

Comment:
Should we use the [caplog fixture](https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture) to verify that the warnings were correctly logged ?